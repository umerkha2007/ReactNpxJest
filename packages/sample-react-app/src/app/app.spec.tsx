import { fireEvent, getByText, render, screen } from '@testing-library/react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import App, { TestableComponent } from './app';

describe('App', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );

    expect(baseElement).toBeTruthy();
  });

  it('should have a greeting as the title', () => {
    const { getByText } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );

    expect(getByText(/Welcome sample-react-app/gi)).toBeTruthy();
  });

  // DOM tests
  it('Facebook link should be present and be clickable', () => {
  const {queryByLabelText, getByLabelText} = render(
    <TestableComponent page="http://facebook.com">Facebook</TestableComponent>,
  );

  expect(queryByLabelText('This is the generated root route')).toBeDefined();

  fireEvent.click(screen.getByText(/Facebook/));

  expect(screen.getByText("Facebook")).toBeDefined();
}); 
  // DOM tests end

  // Snapshot tests
  it('changes the class when hovered', () => {
    const component = renderer.create(
      <TestableComponent page="http://www.facebook.com">Facebook</TestableComponent>,
    );
    let tree = component.toJSON() as ReactTestRendererJSON;
    expect(tree).toMatchSnapshot();
  
    // manually trigger the callback
    renderer.act(() => {
      tree.props.onMouseEnter();
    });
    // re-rendering
    tree = component.toJSON() as ReactTestRendererJSON;
    expect(tree).toMatchSnapshot();
  
    // manually trigger the callback
    renderer.act(() => {
      tree.props.onMouseLeave();
    });
    // re-rendering
    tree = component.toJSON() as ReactTestRendererJSON;;
    expect(tree).toMatchSnapshot();
  });
  // Snapshot tests
});
