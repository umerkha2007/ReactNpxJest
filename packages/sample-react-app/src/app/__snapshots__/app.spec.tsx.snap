// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App changes the class when hovered 1`] = `
Array [
  <h1>
    Status: 
    normal
  </h1>,
  <a
    className="normal"
    href="http://www.facebook.com"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    Facebook
  </a>,
]
`;
